apps/aechannel/src/aesc_utils.erl

```diff
@@ -524,6 +554,36 @@ is_delegatable_tx_type(Type) ->
 delegatable_tx_types() ->
     [slash].
 
+-spec verify_signatures_onchain(aetx_sign:signed_tx(), aec_trees:trees(),
+                                aetx_env:env()) -> ok | {error, atom()}.
+verify_signatures_onchain(SignedTx, Trees, Env) ->
+    verify_signatures_onchain(SignedTx, Trees, Env, []).
+
+verify_signatures_onchain(SignedTx, Trees, Env, CheckedSigners) ->
```

Comment:
I arrived here from clause `awaiting_signature(cast, {?SIGNED, create_tx, Tx} = Msg,` in the FSM.

I have the impression that this function does not prevent not relevant extra ga meta txs - i.e. ga meta tx authenticating accounts not relevant to operation. Though I see that this [is checked on chain](https://github.com/aeternity/aeternity/pull/2354/files#diff-15a4b5418b5f5c5a1f2fb8d8d0d3f619R1330), and FSM waits for tx in chain anyway. So ok.

Though an assertion `aesc_utils:count_authentications(...) =:= 1` would improve readability in code and reduce chance of user(s?) preparing wrong tx. Maybe could be considered as future improvement (outside of this PR).

Reply:
Similarly for code path from `half_signed(cast, {?FND_SIGNED, Msg}, #data{role = initiator,`. Though ok.